napari/viewer.py

```diff
@@ -28,6 +31,8 @@ class Viewer(ViewerModel):
         Whether to show the viewer after instantiation. by default True.
     """
 
+    napari_app_id = 'napari.napari.viewer.' + str(__version__)
```

Comment:
Final change request: let's make this attribute semi-private (`_napari_app_id = `) so that people don't see it when tab-completing with `viewer.<TAB>`. Perhaps there should also be a comment above the attribute: `# set napari.Viewer._napari_app_id to None to avoid overwriting dock icon on Windows`? In fact, is it None or the empty string? I don't know the right way to deal with this, which somewhat defeats the purpose.

Reply:
Done. Also add information, that if you use custom string then it prevent grouping (tested on Windows). 
I add check that prevent setting ctypes if value cast to false.